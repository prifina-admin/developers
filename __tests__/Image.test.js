import React from "react";
import renderer from "react-test-renderer";
import Image from "../src/Image";
import ThemeProvider from "../src/theme/ThemeProvider";

test("Image renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <Image src={"https://picsum.photos/200/200"} />
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Circle Image renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <Image src={"https://picsum.photos/200/200"} shape={"circle"} alt={"Image"} />
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Rounded Image renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <Image
                src={"https://picsum.photos/200/200"}
                shape={"rounded"}
                alt={"Image"}/>
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Square Image renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <Image
            src={"https://picsum.photos/200/200"}
            alt={"Image"}
            shape={"square"}
            width={300}/>
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});
