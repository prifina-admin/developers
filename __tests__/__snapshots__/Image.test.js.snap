// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Image renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <img
    alt="Image"
    className="Image__ImageElement-sc-1qpr0nl-0 kvlogo"
    height="auto"
    src="https://picsum.photos/200/200"
  />
</div>
`;

exports[`Image renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <img
    alt="image"
    className="Image__ImageElement-sc-1qpr0nl-0 bmUaBI"
    height="auto"
    src="https://picsum.photos/200/200"
  />
</div>
`;

exports[`Rounded Image renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <img
    alt="Image"
    className="Image__ImageElement-sc-1qpr0nl-0 kehPyh"
    height="auto"
    src="https://picsum.photos/200/200"
  />
</div>
`;

exports[`Square Image renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <img
    alt="Image"
    className="Image__ImageElement-sc-1qpr0nl-0 ibLAnJ"
    height="auto"
    src="https://picsum.photos/200/200"
    width={300}
  />
</div>
`;
